### PR TITLE
AK: Default move operators for `CircularBuffer`

### PR DESCRIPTION
**Pull request by LucasChollet**
cc @ADKaster

### CODE REVIEW

File: AK/CircularBuffer.h [hunk 1]
```diff
@@ -19,22 +19,8 @@ class CircularBuffer {
...
-            return *this;
-
-        swap(m_buffer, other.m_buffer);
-        swap(m_reading_head, other.m_reading_head);
-        swap(m_used_space, other.m_used_space);
-
-        return *this;
-    }
+    CircularBuffer(CircularBuffer&& other) = default;
+    CircularBuffer& operator=(CircularBuffer&& other) = default;
```
[the remark this answers is not in the document]
Comment (LucasChollet) [reply AK/CircularBuffer.h:23]: It does have a change in behavior, but it won't affect the correctness, as both integers aren't used in the destructor.

I updated the commit message.

[the remark this answers is not in the document]
Comment (ADKaster) [reply AK/CircularBuffer.h:23]: I don't think the distinction in this case matters, does it? The difference in implementations won't be observable.

[the remark this answers is not in the document]
Comment (ADKaster) [reply AK/CircularBuffer.h:23]: Hmm ok looking again at cppreference  https://en.cppreference.com/w/cpp/language/move_constructor

> A trivial move constructor is a constructor that performs the same action as the trivial copy constructor, that is, makes a copy of the object representation as if by [std::memmove](https://en.cppreference.com/w/cpp/string/byte/memmove). All data types compatible with the C language (POD types) are trivially movable.

So the moved-from object will be left with the old values for `m_reading_head`and `m_used_space`, which voilates its invariants. 

But at the end of the day, I don't think we care too much :shrug:

[the remark this answers is not in the document]
Comment (LucasChollet) [reply AK/CircularBuffer.h:23]: AFAICT it doesn't matter, I think that awesomekling wanted me to bring to the fore that there is a conceptual difference.

I wanted a confirmation before resolving the conversation.

**Review by ADKaster — APPROVED** [review]

**Pull request merged**